components/calendar/src/helpers.rs

```diff
@@ -137,19 +137,21 @@ pub fn mod3(x: f64, a: f64, b: f64) -> f64 {
     }
 }
 
-// Arctan of x,y in degrees
-pub fn arctan_degrees(y: f64, x: f64) -> Result<f64, &'static str> {
+// Arctan of y/x in degrees, handling zero cases
+//
+// Returns None if x and y are both 0
+pub fn arctan_degrees(y: f64, x: f64) -> Option<f64> {
     if x == 0.0 && y == 0.0 {
-        Err("Both x and y are zero")
+        None
     } else if x == 0.0 {
         if y > 0.0 {
-            Ok(90.0)
+            Some(90.0)
         } else {
-            Ok(-90.0)
+            Some(-90.0)
         }
     } else {
         let alpha = libm::atan(y / x) * 180.0 / PI;
-        Ok(mod_degrees(if x >= 0.0 { alpha } else { alpha + 180.0 }))
+        Some(mod_degrees(if x >= 0.0 { alpha } else { alpha + 180.0 }))
```

Comment:
This is silly.
1. We are doing a division followed by a one-parameter arctangent;
2. we are branching before it (and making the type awkward, leading to the issues being worked around in the rest of this PR) to avoid atan(NaN), which would not be a problem if we used atan2: we would get ±0 or ±π given two 0s, see IEEE 754-2008 Section 9.2.1;
3. we are branching after it to pick the right quadrant which we lost by using the one-parameter arctangent.

Change this to
```rust
pub fn arctan_degrees(y: f64, x: f64) -> f64 {
        mod_degrees(libm::atan2(y, x) * 180.0 / PI);
}
```
and remove the optionals in the callers accordingly.